core/src/test/java/org/kiwix/kiwixmobile/core/bookmark/viewmodel/BookmarkViewModelTest.kt

```diff
@@ -0,0 +1,157 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.bookmark.viewmodel
+
+import com.jraska.livedata.test
+import io.mockk.clearAllMocks
+import io.mockk.every
+import io.mockk.mockk
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.processors.PublishProcessor
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.schedulers.TestScheduler
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.kiwix.kiwixmobile.core.bookmark.adapter.BookmarkItem
+import org.kiwix.kiwixmobile.core.bookmark.viewmodel.Action.ExitActionModeMenu
+import org.kiwix.kiwixmobile.core.bookmark.viewmodel.Action.ExitBookmarks
+import org.kiwix.kiwixmobile.core.bookmark.viewmodel.Action.Filter
+import org.kiwix.kiwixmobile.core.bookmark.viewmodel.Action.OnItemClick
+import org.kiwix.kiwixmobile.core.bookmark.viewmodel.Action.OnItemLongClick
+import org.kiwix.kiwixmobile.core.bookmark.viewmodel.Action.UpdateBookmarks
+import org.kiwix.kiwixmobile.core.bookmark.viewmodel.Action.UserClickedDeleteButton
+import org.kiwix.kiwixmobile.core.bookmark.viewmodel.Action.UserClickedDeleteSelectedBookmarks
+import org.kiwix.kiwixmobile.core.bookmark.viewmodel.Action.UserClickedShowAllToggle
+import org.kiwix.kiwixmobile.core.bookmark.viewmodel.effects.OpenBookmark
+import org.kiwix.kiwixmobile.core.bookmark.viewmodel.effects.ShowDeleteBookmarksDialog
+import org.kiwix.kiwixmobile.core.bookmark.viewmodel.effects.UpdateAllBookmarksPreference
+import org.kiwix.kiwixmobile.core.dao.NewBookmarksDao
+import org.kiwix.kiwixmobile.core.reader.ZimReaderContainer
+import org.kiwix.kiwixmobile.core.search.viewmodel.effects.Finish
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+import org.kiwix.sharedFunctions.InstantExecutorExtension
+import org.kiwix.sharedFunctions.setScheduler
+
+@ExtendWith(InstantExecutorExtension::class)
+internal class BookmarkViewModelTest {
+  private val bookmarksDao: NewBookmarksDao = mockk()
+  private val zimReaderContainer: ZimReaderContainer = mockk()
+  private val sharedPreferenceUtil: SharedPreferenceUtil = mockk()
+
+  private lateinit var viewModel: BookmarkViewModel
+  private val testScheduler = TestScheduler()
+
+  private val itemsFromDb: PublishProcessor<List<BookmarkItem>> =
+    PublishProcessor.create()
+
+  init {
+    setScheduler(testScheduler)
```

Comment:
you don't need testScheduler so delete.

you do however need the IO scheduler to run on `trampoline` ie sequentially so you can just say

`setScheduler(Schedulers.trampoline())`